# IMPORT LIBRARY

In [ ]:
install.packages("rstatix")
library(rstatix)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

also installing the dependencies ‘colorspace’, ‘fracdiff’, ‘lmtest’, ‘timeDate’, ‘urca’, ‘zoo’, ‘RcppArmadillo’, ‘cowplot’, ‘Deriv’, ‘forecast’, ‘microbenchmark’, ‘rbibutils’, ‘numDeriv’, ‘doBy’, ‘SparseM’, ‘MatrixModels’, ‘Rdpack’, ‘minqa’, ‘nloptr’, ‘reformulas’, ‘RcppEigen’, ‘carData’, ‘abind’, ‘Formula’, ‘pbkrtest’, ‘quantreg’, ‘lme4’, ‘corrplot’, ‘car’



Attaching package: ‘rstatix’


The following object is masked from ‘package:stats’:

    filter




In [ ]:
install.packages("tidyverse")
library(tidyverse)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.2.0     ✔ readr     2.2.0
✔ forcats   1.0.1     ✔ stringr   1.6.0
✔ ggplot2   4.0.2     ✔ tibble    3.3.1
✔ lubridate 1.9.5     ✔ tidyr     1.3.2
✔ purrr     1.2.1     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks rstatix::filter(), stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [ ]:
install.packages("ggplot2")
library(ggplot2)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [ ]:
install.packages("ggpubr")
library(ggpubr)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

also installing the dependencies ‘ggrepel’, ‘ggsci’, ‘ggsignif’, ‘gridExtra’, ‘polynom’




In [ ]:
install.packages("readxl")
library(readxl)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [ ]:
install.packages("dplyr")
library(dplyr)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [ ]:
install.packages("readxlsx")
library(readxlsx)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Warning message:
“package ‘readxlsx’ is not available for this version of R

A version of this package for your version of R might be available elsewhere,
see the ideas at
https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages”


ERROR: Error in library(readxlsx): there is no package called ‘readxlsx’


# COBA UJI MANN WHITNEY U
**UNTUK MENGETAHUI APAKAH ADA PERBEDAAN RATA2 USIA KAWIN PERTAMA DI KABUPATEN DAN DI KOTA**

In [ ]:
data = read_xlsx("/content/Data ADT.xlsx", sheet = 7)
data

usia_kawin,wilayah
<dbl>,<chr>
20.05,Kabupaten
20.27,Kabupaten
20.19,Kabupaten
20.46,Kabupaten
20.51,Kabupaten
20.21,Kabupaten
20.41,Kabupaten
20.35,Kabupaten
20.52,Kabupaten


In [ ]:
wilcox.test(usia_kawin ~ wilayah, data = data)

Warning message in wilcox.test.default(x = DATA[[1L]], y = DATA[[2L]], ...):
“cannot compute exact p-value with ties”



	Wilcoxon rank sum test with continuity correction

data:  usia_kawin by wilayah
W = 0, p-value = 6.786e-08
alternative hypothesis: true location shift is not equal to 0


P VALUE = 6.786e-08 = 0.00000006786

ADA PERBEDAAN SIGNIFIKAN USIA RATA2 KAWIN PERTAMA DI KOTA DAN DI KABUPATEN


# UJI WILCOXON UNTUK MENGETAHUI APAKAH ADA PERBEDAAN USIA RATA2 KAWIN PERTAMA PADA TAHUN 2019  DAN TAHUN 2023

### DI KABUPATEN

In [ ]:
data = read_xlsx("/content/Data ADT.xlsx",sheet = 5)
data

KABUPATEN,2019,2020,2021,2022,2023
<chr>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>
Kabupaten Pandeglang,20.05305,20.27255,20.19280,20.45710,20.51000
Kabupaten Lebak,20.20530,20.40555,20.35010,20.51980,20.61260
Kabupaten Tangerang,21.17650,21.37670,21.31445,21.49695,21.55675
Kabupaten Serang,20.96815,21.16435,21.08025,21.24345,21.31475


In [ ]:
data_kabupaten = data
data_kabupaten

KABUPATEN,2019,2020,2021,2022,2023
<chr>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>
Kabupaten Pandeglang,20.05305,20.27255,20.19280,20.45710,20.51000
Kabupaten Lebak,20.20530,20.40555,20.35010,20.51980,20.61260
Kabupaten Tangerang,21.17650,21.37670,21.31445,21.49695,21.55675
Kabupaten Serang,20.96815,21.16435,21.08025,21.24345,21.31475


In [ ]:
wilcox.test(data_kabupaten$`2019`, data_kabupaten$`2023`,
            paired = TRUE,
            alternative = "two.sided",  # bisa juga "greater" / "less"
            conf.int = TRUE)

Warning message in wilcox.test.default(data_kabupaten$`2019`, data_kabupaten$`2023`, :
“requested conf.level not achievable”



	Wilcoxon signed rank exact test

data:  data_kabupaten$`2019` and data_kabupaten$`2023`
V = 0, p-value = 0.125
alternative hypothesis: true location shift is not equal to 0
88 percent confidence interval:
 -0.45695 -0.34660
sample estimates:
(pseudo)median 
     -0.397775 


p < 0.05 → ada perbedaan signifikan

p ≥ 0.05 → tidak ada perbedaan signifikan


KESIMPULAN = NILAI P VALUE = 0.125 > 0,05 ARTINYA tidak ada perbedaan signifikan rata2 usia kawin di tahun 2019 dan 2023 di kabupaten

### DI KOTA

In [ ]:
data = read_xlsx("/content/Data ADT.xlsx", sheet = 6)
data

data_kota = data

KABUPATEN,2019,2020,2021,2022,2023
<chr>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>
Kota Tangerang,22.28315,22.40845,22.35085,22.51450,22.57330
Kota Cilegon,21.77245,21.93735,21.86490,22.03460,22.09340
Kota Serang,21.62570,21.79060,21.71815,21.89885,21.95765
Kota Tangerang Selatan,22.70360,22.82450,22.76510,22.91515,22.97395


In [ ]:
wilcox.test(data_kota$`2019`, data_kota$`2023`,
            paired = TRUE,
            alternative = "two.sided",  # bisa juga "greater" / "less"
            conf.int = TRUE)

Warning message in wilcox.test.default(data_kota$`2019`, data_kota$`2023`, paired = TRUE, :
“requested conf.level not achievable”



	Wilcoxon signed rank exact test

data:  data_kota$`2019` and data_kota$`2023`
V = 0, p-value = 0.125
alternative hypothesis: true location shift is not equal to 0
88 percent confidence interval:
 -0.33195 -0.27035
sample estimates:
(pseudo)median 
      -0.30335 


p < 0.05 → ada perbedaan signifikan

p ≥ 0.05 → tidak ada perbedaan signifikan


KESIMPULAN = NILAI P VALUE = 0.125 > 0,05 ARTINYA tidak ada perbedaan signifikan rata2 usia kawin di tahun 2019 dan 2023 di kota

### KABUPATEN DAN KOTA

In [ ]:
data = read_xlsx("/content/Data ADT.xlsx" , sheet =  4)
data

KABUPATEN DAN KOTA,2019,2020,2021,2022,2023
<chr>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>
Kabupaten Pandeglang,20.05305,20.27255,20.19280,20.45710,20.51000
Kabupaten Lebak,20.20530,20.40555,20.35010,20.51980,20.61260
Kabupaten Tangerang,21.17650,21.37670,21.31445,21.49695,21.55675
Kabupaten Serang,20.96815,21.16435,21.08025,21.24345,21.31475
Kota Tangerang,22.28315,22.40845,22.35085,22.51450,22.57330
Kota Cilegon,21.77245,21.93735,21.86490,22.03460,22.09340
Kota Serang,21.62570,21.79060,21.71815,21.89885,21.95765
Kota Tangerang Selatan,22.70360,22.82450,22.76510,22.91515,22.97395


In [ ]:
data$`2019` = as.numeric(data$`2019`)
data$`2023` = as.numeric(data$`2023`)

In [ ]:
wilcox.test(data$`2019`,data$`2023`,
            paired = TRUE,
            alternative = "two.sided",
            conf.int = TRUE)


	Wilcoxon signed rank exact test

data:  data$`2019` and data$`2023`
V = 0, p-value = 0.007813
alternative hypothesis: true location shift is not equal to 0
95 percent confidence interval:
 -0.40730 -0.29565
sample estimates:
(pseudo)median 
    -0.3476625 


p < 0.05 → ada perbedaan signifikan

p ≥ 0.05 → tidak ada perbedaan signifikan


KESIMPULAN = NILAI P VALUE = 0.007813 < 0,05 ARTINYA ada perbedaan signifikan rata2 usia kawin di tahun 2019 dan 2023 di kota dan di kabupaten

# UJI WILCOXON DI UNTUK MENGETAHUI APAKAH ADA PERBEDAAN RATA2 KAWIN PERTAMA DI KABUPATEN DAN KOTA  DARI 2019 SAMPAI 2023

In [ ]:
data = read_xlsx("/content/Data ADT.xlsx", sheet = 8)
data

KABUPATEN,KOTA
<dbl>,<dbl>
22.28315,22.28315
22.40845,22.40845
22.35085,22.35085
22.51450,22.51450
22.57330,22.57330
20.20530,21.77245
20.40555,21.93735
20.35010,21.86490
20.51980,22.03460


In [ ]:
data$KABUPATEN = as.numeric(data$KABUPATEN)
data$KOTA = as.numeric(data$KOTA)

In [ ]:
wilcox.test(data$KABUPATEN,data$KOTA,
            paired = TRUE,
            alternative = "two.sided",  # bisa juga "greater" / "less"
            conf.int = TRUE)

Warning message in wilcox.test.default(data$KABUPATEN, data$KOTA, paired = TRUE, :
“cannot compute exact p-value with zeroes”
Warning message in wilcox.test.default(data$KABUPATEN, data$KOTA, paired = TRUE, :
“cannot compute exact confidence interval with zeroes”



	Wilcoxon signed rank test with continuity correction

data:  data$KABUPATEN and data$KOTA
V = 0, p-value = 0.0007265
alternative hypothesis: true location shift is not equal to 0
95 percent confidence interval:
 -1.6080770 -0.9593334
sample estimates:
(pseudo)median 
     -1.067541 


In [ ]:
wilcox.test(data$KABUPATEN, data$KOTA,
            paired = TRUE,
            exact = FALSE,
            conf.int = TRUE)


	Wilcoxon signed rank test with continuity correction

data:  data$KABUPATEN and data$KOTA
V = 0, p-value = 0.0007265
alternative hypothesis: true location shift is not equal to 0
95 percent confidence interval:
 -1.6080770 -0.9593334
sample estimates:
(pseudo)median 
     -1.067541 


p < 0.05 → ada perbedaan signifikan

p ≥ 0.05 → tidak ada perbedaan signifikan


KESIMPULAN = NILAI P VALUE = 0.0007265 < 0,05

ARTINYA  ada perbedaan signifikan rata2 usia kawin di tahun 2019 SAMPAI 2023 di kabupaten dan kota